lightning/src/chain/keysinterface.rs

```diff
@@ -49,43 +49,66 @@ pub enum SpendableOutputDescriptor {
 		output: TxOut,
 	},
 	/// An output to a P2WSH script which can be spent with a single signature after a CSV delay.
-	/// The private key which should be used to sign the transaction is provided, as well as the
-	/// full witness redeemScript which is hashed in the output script_pubkey.
+	///
 	/// The witness in the spending input should be:
-	/// <BIP 143 signature generated with the given key> <empty vector> (MINIMALIF standard rule)
-	/// <witness_script as provided>
-	/// Note that the nSequence field in the input must be set to_self_delay (which corresponds to
-	/// the transaction not being broadcastable until at least to_self_delay blocks after the input
-	/// confirms).
+	/// <BIP 143 signature> <empty vector> (MINIMALIF standard rule) <provided witnessScript>
+	///
+	/// Note that the nSequence field in the spending input must be set to to_self_delay
+	/// (which means the transaction is not broadcastable until at least to_self_delay
+	/// blocks after the outpoint confirms).
+	///
 	/// These are generally the result of a "revocable" output to us, spendable only by us unless
-	/// it is an output from us having broadcast an old state (which should never happen).
+	/// it is an output from an old state which we broadcast (which should never happen).
+	///
+	/// To derive the delayed_payment key which is used to sign for this input, you must pass the
+	/// local delayed_payment_base_key (ie the private key which corresponds to the pubkey in
+	/// ChannelKeys::pubkeys().delayed_payment_basepoint) and the provided per_commitment_point to
+	/// chan_utils::derive_private_key. The public key can be generated without the secret key
+	/// using chan_utils::derive_public_key and only the delayed_payment_basepoint which appears in
+	/// ChannelKeys::pubkeys().
+	///
+	/// To derive the revocation_pubkey which is used in the witness script generation, you must
+	/// pass the remote revocation_basepoint (which appears in the call to
+	/// ChannelKeys::set_remote_channel_pubkeys) and the provided per_commitment point to
```

Comment:
I realize my changed wording is slightly confusing, because we actually do provide it here, and its awkward af to rely on the user having stored the remote channel keys in a way we can access them by looking up with the derivation keys.
```suggestion
	/// To re-derive the remote_revocation_pubkey provided here (which is used in the witness
	/// script generation), you must pass the remote revocation_basepoint (which appears in the
	/// call to ChannelKeys::set_remote_channel_pubkeys) and the provided per_commitment point to
```